R/reorder.R

```diff
@@ -68,6 +69,30 @@ fct_reorder2 <- function(.f, .x, .y, .fun = last2, ..., .desc = TRUE) {
   lvls_reorder(.f, order(summary, decreasing = .desc))
 }
 
+#' Reorder factor levels by sorting along multiple variables
+#' 
+#' @param .f A factor (or character vector).
+#' @param method Passed to \code{base::order()}
+#' @param ordered Passed to \code{\link{fct_inorder}()}
+#' @inheritDotParams base::order -method
+#' @family Reordering
+#' @examples
+#' A <- c(3, 3, 2, 1)
+#' B <- c("A", "B", "C", "D")
+#' fct_reordern(.f = c("A", "B", "C", "D"), A, B)
+#' fct_reordern(.f = c("A", "B", "C", "D"), A, B, decreasing = TRUE)
+#' fct_reordern(.f = c("A", "B", "C", "D"), A, B, decreasing = c(FALSE, TRUE))
+#' @export
+fct_reordern <- function(.f, ..., ordered = NA, method="radix") {
+  f <- check_factor(.f)
+  new_order <- base::order(..., method=method)
+  f_sorted <- f[new_order]
+  # Don't keep the value order for new_f as they will be inaccurately sorted.
+  # Just keep the levels and ordered state.
+  new_f <- fct_inorder(f_sorted, ordered = ordered)
```

Comment:
I don't understand the logic here. Can you please use one of the existing reordering functions (e.g. `forcats::lvls_reorder()`)?

Reply:
Ah, I didn't know of `lvls_reorder()`.  Simplified.